### PR TITLE
RET-4926: ResponseReceivedCounter for WA

### PR DESCRIPTION
**Pull request by fergusoshea**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RET-4926

### Change description ###
Update Counter for WA

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -336,7 +336,7 @@ dependencyManagement {
 dependencies {
 
     implementation group: 'com.github.hmcts', name: 'et-common', version:'1.5.7'
-    implementation group: 'com.github.hmcts', name: 'et-data-model', version:'3.6.6'
+    implementation group: 'com.github.hmcts', name: 'et-data-model', version:'3.6.8-beta'
```
Comment (marcjudge) [inline build.gradle:339]: I think you're okay to use a non beta release

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 2]
```diff
@@ -267,11 +269,36 @@ public void setEt3ResponseDueDate(CaseData caseData) {
...
+            updateResponseReceivedCounter(caseData.getRespondentCollection());
+        }
+    }
+
+    private void updateResponseReceivedCounter(List<RespondentSumTypeItem> respondentCollection) {
+        RespondentSumType firstRespondent = respondentCollection.get(0).getValue();
+        if (YES.equals(firstRespondent.getResponseReceived())) {
+            firstRespondent.setResponseReceivedCount(
+                    StringUtils.isBlank(firstRespondent.getResponseReceivedCount())
+                            ? "1"
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:291]: Avoid use of a magic number and use a descriptive constant

File: build.gradle (hunk 1 above)
Comment (fergusoshea) [reply build.gradle]: I have created the 3.6.8 Release

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java (hunk 2 above)
Comment (fergusoshea) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java]: Excellent point, updated.

**Review by marcjudge — APPROVED** [review]

**Review by JackReevies — APPROVED** [review]

**Review by ss-evoco — APPROVED** [review]

**Pull request merged**